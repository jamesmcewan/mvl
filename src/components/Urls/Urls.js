import React from 'react';
import { isArray } from 'util';

const Urls = ({ urls }) => (
  <ul>
    {isArray(urls) &&
      urls.map(url => (
        <li key={`url-${url.url}`}>
          <a href={url.url}>{url.type}</a>
        </li>
      ))}
  </ul>
);

Urls.defaultProps = {
  urls: [
    {
      url: '',
      type: '',
    },
  ],
};

export default Urls;
