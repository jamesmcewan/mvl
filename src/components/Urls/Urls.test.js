import React from 'react';
import { mount } from 'enzyme';
import Urls from './Urls';
import 'jest-styled-components';

describe('Urls', () => {
  it('should render without crashing', () => {
    const wrapper = mount(<Urls />);
    expect(wrapper).toHaveLength(1);
  });

  it('should render the correct JSX', () => {
    const wrapper = mount(<Urls />);
    expect(wrapper).toMatchSnapshot();
  });
});
