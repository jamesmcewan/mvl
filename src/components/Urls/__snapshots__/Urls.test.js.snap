// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Urls should render the correct JSX 1`] = `
<Urls
  urls={
    Array [
      Object {
        "type": "",
        "url": "",
      },
    ]
  }
>
  <ul>
    <li
      key="url-"
    >
      <a
        href=""
      />
    </li>
  </ul>
</Urls>
`;
